### PR TITLE
Non-resizable modular GUI with canvas zoom

### LINKED ISSUE #6: Implement Recursive Fractal Drawing with Tkinter
Implement a recursive fractal drawing feature using Tkinter. The fractal should be generated based on user-defined parameters and displayed in a GUI.  

### 1. Recursive Fractal Implementation: 
- Implement the fractal generation using a recursive algorithm.
- Ensure the fractal can be drawn to a specified depth.

#### Parameters of the main (modifiable) fractal tree:
_Constant_: 
  - Start Coordinates (start_x, start_y): The starting point of the fractal—the middle of the x axis and the bottom of the y axis.
  - Angle: The initial angle of the trunk of the fractal tree—it is set constant at -90 degrees.
 
 _Modifiable:_
  - Depth: The maximum depth of recursion. Controls the complexity of the fractal.
  - Length: The initial length of the fractal branches.

  - Number of Children (num_children): The number of branches each node will have.
  - Length Scale (length_scale): The ratio of the length of child branches to the parent branch.
  - Delta Angle (delta_angle): The angle difference between consecutive branches.
  - Color (base_color and leaves_color)

### 2. Tkinter Integration (GUI):

- [x] Create a Tkinter window to display the fractal.
- [ ] Create sliders for each parameter specified as adjustable. They let the user manipulate the parameters, which should translate to the look of the fractal.
- [ ] Tie the scales to the fractal generating functions to modify the fractal as the parameters get adjusted.
- [ ]  _Optional:  Implement 2 color sliders either as buttons to choose a color or as slider to choose color hue, possibly with a visual representation of the values as a Label (perhaps).
Let the user choose the colors with Tkinter's `colorchooser` to or [use the sliders to choose the hue](https://stackoverflow.com/a/30688948/9548903) with [HSL or  HSV color map](https://en.wikipedia.org/wiki/HSL_and_HSV)  (the saturation and lightness/value would be predefined to provide best contrast)._

## Other Tasks
- [x] Implement the recursive fractal generation algorithm.
- [x] Create a Tkinter GUI with input fields for parameters and canvas with zoom feature.
- [x] Implement the drawing logic using the Canvas widget.
- [x] Write the logic to modify the fractal tree based on parameter change (by sliders) during runtime.
- [ ] Test the application to ensure it works as expected—write basic unit tests and do manual testing.

### PR DESCRIPTION
**Pull request by ola-sk**
## New features introduced:
- Improved Sliders
- Zoom functionality
- Refactored and modularised code
- Module and function documentation

## Slider Functionality:
`gui/sliders.py`: Added functions to create sliders, generate column sequences, and populate sliders within a specified frame.

## Zoom Functionality:
`gui/zoom.py`: Implemented functions for panning and zooming on a Tkinter canvas, and a function to bind these events to the canvas.
`sandbox/draw_on_canvas.py`: Added functionality to draw squares on the canvas, with parameters for the number of squares, size, and color. Mostly result of the tests of the zoom functionality.

 ## Window creation:

`gui/gui_init.py`: Introduced the _initialise_gui_ function to create a non-resizable Tkinter window with a specified title, width, and height.
The _initialise_gui_ function gets its size in int as a parameter, which lets to set the canvas relative to the window size from the `mainwindow.py`.

## Package definition
`gui/__init__.py`: Added initialization module for the GUI package, importing and exposing functions for initializing the GUI, creating sliders, and binding canvas zoom events. That lets us call: 

```python
from gui import initialise_gui, populate_sliders, bind_canvas_zoom_events

```

`sandbox/__init__.py`: Imported and exposed the _draw_squares_ function from the `draw_on_canvas.py` module. This file is an initialization module for the Sandbox package, which holds things like experimental functions for drawing on canvas, e.g. to easier visually inspect features like zooming in/out on the canvas.

## Main Application Integration:
`mainwindow.py`: The main application module uses modular code to  initialize the main window, set up the canvas, add sliders and bind zoom events. Integrated the _initialise_gui_, _populate_sliders_, and _bind_canvas_zoom_events_ functions.

## Screenshot
image

### CODE REVIEW

File: gui/sliders.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+"""
+Main module for the application.
+
+This module initializes the main application window, sets up the canvas,
+adds sliders for various parameters, and binds events for canvas zooming.
```
Comment (ola-sk) [inline gui/sliders.py]: That's wrong description! I must have mistaken the modules. It is a description same as for the mainwindow. Will fix before the merge.

File: gui/zoom.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def do_pan(event: tk.Event):
+    """Drags the canvas to the new position."""
+    canvas: tk.Canvas = event.widget
+    canvas.scan_dragto(event.x, event.y, gain=1)
+
+
+def zoom(event: tk.Event):
+    """Zooms in or out on the canvas."""
```
Comment (Iv-Tatiana) [inline gui/zoom.py]: With the adjustment for pan

File: mainwindow.py [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
 
-# Creating main application window
-window=tk.Tk()
-window.title("Fractal Matching Game") 
-window.geometry("600x800") # Set to potrait orientation
-window.resizable(False,False) # Making window non resizable
+from gui import (initialise_gui,
+                 populate_sliders,
+                 bind_canvas_zoom_events)
+from sandbox import draw_squares
```
Comment (Iv-Tatiana) [inline mainwindow.py:13]: I like that you moved draw_squares into a separate file. I didn't think of doing it, but I also didn't plan to make this an official part of the project. Thanks for such a nice treatment =)

File: mainwindow.py [hunk 4]
```diff
@@ -1,10 +1,43 @@
...
+canvas_padding_x = 0
+canvas = tk.Canvas(
+    window, bg="white",
+    width=window_width - 2 * canvas_padding_x,
+    height=535
+)
+canvas.grid(
+    row=0, column=0, columnspan=2,
+    pady=(0, 0),
+    padx=(canvas_padding_x, canvas_padding_x))
```
Comment (Iv-Tatiana) [inline mainwindow.py]: it looks like the closing ")" should be moved to the next line

**Review by Iv-Tatiana — CHANGES_REQUESTED** [review]

File: gui/zoom.py (hunk 2 above)
Comment (ola-sk) [reply gui/zoom.py]: Thanks for pointing it out. This was left out as original. I'll update the docstring with the upcoming commit.

File: mainwindow.py (hunk 3 above)
Comment (ola-sk) [reply mainwindow.py:13]: I loved them ❤ I guess it needs to removed in the next iteration though, since the tree will be used instead.

**Review by bluedinos — DISMISSED** [review]

File: gui/zoom.py (hunk 2 above)
Comment (ola-sk) [reply gui/zoom.py]: @Iv-Tatiana Can you check out whether the documentation is clear enough now (and correct)?

Comment (Iv-Tatiana) [reply gui/zoom.py]: Yes! Thank you!

**Review by Iv-Tatiana — APPROVED** [review]
I reviewed the changes, and everything looks good. Approved!

**Pull request merged**